### PR TITLE
Development meetingschatflow

### PR DESCRIPTION
**Pull request by tobiaschielens**
### Description

Giving the ability to deploy a meeting solution. 

### Changes

Implemented meetings.

### Related Issues

https://github.com/threefoldtech/freeflowconnect_frontend/issues/238

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings

### CODE REVIEW

File: jumpscale/packages/marketplace/chats/meetings.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    steps = [
+        "get_solution_name",
+        "meetings_info",
+        "infrastructure_setup",
+        "reservation",
+        "initializing",
+        "success",
+    ]
+
+    query = {"cru": 2, "mru": 2, "sru": 0}
```
Comment (m-motawea) [inline jumpscale/packages/marketplace/chats/meetings.py:19]: the query is wrong. it doesn't include the nginx container resources which makes the expiration incorrect.
should add 1cru, 1 mru, 0.25 sru, and handle these differences in during i the deployment of the container

File: jumpscale/packages/marketplace/chats/meetings.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            solution_uuid=self.solution_id,
+            **self.solution_metadata,
+        )
+
+        success = deployer.wait_workload(_id, self)
+        if not success:
+            raise DeploymentFailed(
+                f"Failed to create subdomain {self.domain} on gateway"
+                f" {self.gateway.node_id} {_id}. The resources you paid for will be re-used in your upcoming deployments."
+            )
```
Comment (m-motawea) [inline jumpscale/packages/marketplace/chats/meetings.py:54]: must add `wid=_id` in the exception, otherwise, it will fail in retries

**Review by m-motawea — CHANGES_REQUESTED** [review]

**Comment by m-motawea** [comment]
also please add patch class to be used in tests for this solution.
you can find the details on how to create one here https://github.com/threefoldtech/js-sdk/blob/development/solutions_automation/README.md

File: jumpscale/packages/marketplace/frontend/data.js [hunk 3]
```diff
@@ -97,6 +97,14 @@ const SECTIONS = {
         helpLink: "https://now10.threefold.io/docs",
         description: "Discourse is an open source Internet forum and mailing list management software application built to educate members about civil community engagement."
       },
+      meetings: {
+        name: "Meetings",
+        type: "meetings",
+        image: "./assets/mattermost.png",
```
Comment (m-motawea) [inline jumpscale/packages/marketplace/frontend/data.js:103]: should use the correct logo for the solution.

File: jumpscale/packages/marketplace/chats/meetings.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            cpu=self.query["cru"],
+            memory=self.query["mru"] * 1024,
+            disk_size=self.query["sru"] * 1024,
+            interactive=False,
+            entrypoint="/entrypoint.sh",
+            public_ipv6=True,
+            solution_uuid=self.solution_id,
+            **self.solution_metadata,
+        )
+        # import ipdb; ipdb.set_trace()
```
Comment (m-motawea) [inline jumpscale/packages/marketplace/chats/meetings.py:73]: should be removed

**Review by m-motawea — CHANGES_REQUESTED** [review]

File: jumpscale/sals/marketplace/solutions.py [hunk 5]
```diff
@@ -45,6 +45,9 @@ def list_gitea_solutions(self, username, next_action=NextAction.DEPLOY, sync=Tru
     def list_mattermost_solutions(self, username, next_action=NextAction.DEPLOY, sync=True):
         return self._list_proxied_solution("mattermost", next_action, sync, "nginx", owner=username)
 
+    def list_meetings_solutions(self, username, next_action=NextAction.DEPLOY, sync=True):
+        return self._list_proxied_solution("meetings", next_action, sync, owner=username)
```
Comment (waleedhammam) [inline jumpscale/sals/marketplace/solutions.py:49]: We should specify the listing type as the default is trc, in the solution it's nginx

File: jumpscale/packages/marketplace/chats/meetings.py (hunk 1 above)
Comment (waleedhammam) [reply jumpscale/packages/marketplace/chats/meetings.py:19]: keep in mind we should distinguish between the total query and the container resources
See: https://github.com/threefoldtech/js-sdk/blob/development_backupcryptpad/jumpscale/packages/tfgrid_solutions/chats/cryptpad_advanced.py#L24

https://github.com/threefoldtech/js-sdk/blob/development_backupcryptpad/jumpscale/packages/tfgrid_solutions/chats/cryptpad_advanced.py#L39

https://github.com/threefoldtech/js-sdk/blob/development_backupcryptpad/jumpscale/packages/tfgrid_solutions/chats/cryptpad_advanced.py#L131

**Review by waleedhammam — CHANGES_REQUESTED** [review]

File: jumpscale/packages/marketplace/frontend/data.js [hunk 6]
```diff
@@ -97,6 +97,14 @@ const SECTIONS = {
         helpLink: "https://now10.threefold.io/docs",
         description: "Discourse is an open source Internet forum and mailing list management software application built to educate members about civil community engagement."
       },
+      meetings: {
+        name: "Meetings",
+        type: "meetings",
+        image: "./assets/meetings.png",
+        disable: false,
+        helpLink: "TODO",
+        description: "Decentralized meetings software."
```
Comment (dmahmouali) [inline jumpscale/packages/marketplace/frontend/data.js:106]: very short description of the app.

File: jumpscale/packages/marketplace/frontend/data.js [hunk 7]
```diff
@@ -97,6 +97,14 @@ const SECTIONS = {
         helpLink: "https://now10.threefold.io/docs",
         description: "Discourse is an open source Internet forum and mailing list management software application built to educate members about civil community engagement."
       },
+      meetings: {
+        name: "Meetings",
+        type: "meetings",
+        image: "./assets/meetings.png",
+        disable: false,
+        helpLink: "https://now10.threefold.io/docs",
```
Comment (xmonader) [inline jumpscale/packages/marketplace/frontend/data.js:105]: shouldn't it be https://now10.threefold.io/docs/video-conf/ ?

File: jumpscale/packages/marketplace/chats/meetings.py [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    steps = [
+        "get_solution_name",
+        "meetings_info",
+        "infrastructure_setup",
+        "reservation",
+        "initializing",
+        "success",
+    ]
+
+    query = {"cru": 1, "mru": 1, "sru": 0.5}
```
Comment (waleedhammam) [inline jumpscale/packages/marketplace/chats/meetings.py:19]: this is the nginx query only, we should specify the workload resources + the nginx container resources.

File: jumpscale/packages/marketplace/frontend/data.js (hunk 7 above)
Comment (tobiaschielens) [reply jumpscale/packages/marketplace/frontend/data.js:105]: Doesn't matter for me, you can change it to whatever suits best for you

Comment (xmonader) [reply jumpscale/packages/marketplace/frontend/data.js:105]: now10 is the commercial content, all of the apps listed in the marketplace should link to the commercial content, please update it to the meeting entry in now10

File: jumpscale/packages/marketplace/chats/meetings.py (hunk 8 above)
Comment (tobiaschielens) [reply jumpscale/packages/marketplace/chats/meetings.py:19]: This is the exact same as the mattermost solution from which I copied a lot. Please explain further. https://github.com/threefoldtech/js-sdk/blob/b6efd79548528e4d2f82c0be5a84093a74ea5716/jumpscale/packages/marketplace/chats/mattermost.py#L19

Comment (waleedhammam) [reply jumpscale/packages/marketplace/chats/meetings.py:19]: just like that comment https://github.com/threefoldtech/js-sdk/pull/1379#discussion_r501928635

we specify the query ( that's the total resources, main container + the nginx container )
then we have the main container resources, which we will use in the _deploy method

Please check the threebot example or this one https://github.com/threefoldtech/js-sdk/blob/490f800386a44ae1c8e4c668d6a2dfa2bd77fc0c/jumpscale/packages/marketplace/chats/taiga.py#L24

**Comment by waleedhammam** [comment]
Can we sync what is the status for this PR ?

**Review by waleedhammam — APPROVED** [review]
Screenshot from 2020-10-26 11-04-30
Screenshot from 2020-10-26 11-04-13

File: solutions_automation/marketplace/meetings.py [hunk 9]
```diff
@@ -0,0 +1,6 @@
+from jumpscale.packages.marketplace.chats.meetings import MeetingsDeploy
+from utils.gedispatch import GedisChatBotPatch
```
Comment (m-motawea) [inline solutions_automation/marketplace/meetings.py:2]: please change this import to `from solutions_automation.utils.gedispatch import GedisChatBotPatch`

**Review by m-motawea — APPROVED** [review]

**Pull request merged**